packages/flutter/lib/src/rendering/table.dart

```diff
@@ -0,0 +1,784 @@
+class IntrinsicColumnWidth extends ColumnWidth { 
+  const IntrinsicColumnWidth();
+
+  @override
+  double minIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    double result = 0.0;
+    for (RenderBox cell in cells)
+      result = math.max(result, cell.getMinIntrinsicWidth(const BoxConstraints()));
+    return result;
+  }
+
+  @override
+  double maxIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    double result = 0.0;
+    for (RenderBox cell in cells)
+      result = math.max(result, cell.getMaxIntrinsicWidth(const BoxConstraints()));
+    return result;
+  }
+}
+
+/// Sizes the column to a specific number of pixels.
+///
+/// This is the cheapest way to size a column.
+class FixedColumnWidth extends ColumnWidth {
+  const FixedColumnWidth(this.value);
+  final double value;
+
+  @override
+  double minIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    return value;
+  }
+
+  @override
+  double maxIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    return value;
+  }
+}
+
+/// Sizes the column to a fraction of the table's constraints' maxWidth.
+///
+/// This is a cheap way to size a column.
+class FractionColumnWidth extends ColumnWidth {
+  const FractionColumnWidth(this.value);
+  final double value;
+
+  @override
+  double minIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    if (!containerWidth.isFinite)
+      return 0.0;
+    return value * containerWidth;
+  }
+
+  @override
+  double maxIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    if (!containerWidth.isFinite)
+      return 0.0;
+    return value * containerWidth;
+  }
+}
+
+/// Sizes the column by taking a part of the remaining space once all
+/// the other columns have been laid out.
+///
+/// For example, if two columns have FlexColumnWidth(), then half the
+/// space will go to one and half the space will go to the other.
+///
+/// This is a cheap way to size a column.
+class FlexColumnWidth extends ColumnWidth {
+  const FlexColumnWidth([this.value = 1.0]);
+  final double value;
+
+  @override
+  double minIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    return 0.0;
+  }
+
+  @override
+  double maxIntrinsicWidth(Iterable<RenderBox> cells, double containerWidth) {
+    return 0.0;
+  }
+
+  @override
+  double flex(Iterable<RenderBox> cells) {
+    return value;
+  }
+}
+
+/// Sizes the column such that it is the size that is the maximum of
+/// two column width specifications.
+///
+/// For example, to have a column be 10% of the container width or
+/// 100px, whichever is bigger, you could use:
+///
+///     const MaxColumnWidth(const FixedColumnWidth(100.0), FractionColumnWidth(0.1))
+///
+/// Both specifications are evaluated, so if either specification is
+/// expensive, so is this.
+class MaxColumnWidth extends ColumnWidth {
+  const MaxColumnWidth(this.a, this.b);
```

Comment:
to be the same as `math.max()`, because most use cases will only need two (the ideal width and the limiting width) and because it's inefficient enough as it is, no need to allocate a whole list and encourage people to give multiple values.

But I don't really feel strongly about this; if you think it should be a list I'm happy to change it. Let me know.